File: projecteuler/PE_001/Python/PE_001_naive.py
import sys


def is_divisible(num: int, divisors: list) -> bool:
    return any(num % d == 0 for d in divisors)


def yield_divisible(numbers: list, divisors: list) -> int:
    for number in numbers:
        if is_divisible(number, divisors):
            yield number


def PE_001_naive(start: int, stop: int, divisors: list) -> int:
    numbers = range(start, stop)
    return sum(yield_divisible(numbers, divisors))


if __name__ == "__main__":

    if len(sys.argv) == 2:
        args = sys.argv[1].replace(" ", "").split("|")
        start, stop, divisors = int(args[0]), int(args[1]), eval(args[2])
    else:
        start, stop, divisors = [1, 1000, [3, 5]]

    print(PE_001_naive(start, stop, divisors))

File: projecteuler/PE_001/Kotlin/PE_001_naive.kt
val divisible:(Int, Int) -> Boolean = { num, div -> num % div == 0 }

fun isDivisible (number: Int, divisors: List<Int>): Boolean { 
  return divisors.any{ divisible(number, it) }
}

fun projectEuler001 (start: Int, stop: Int, divisors: List<Int>): Int {
  return (start..stop-1)
         .filter{ isDivisible(it, divisors) }
         .sum()
}

fun main() {
  val divisors = listOf(3, 5)
  val start = 1
  val stop = 1000
  println(projectEuler001(start, stop, divisors))
}
